cmd/report/template.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2021, SailPoint Technologies, Inc. All rights reserved.
+package report
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	v3 "github.com/sailpoint-oss/golang-sdk/v3"
+	"github.com/sailpoint-oss/sailpoint-cli/internal/config"
+	"github.com/sailpoint-oss/sailpoint-cli/internal/log"
+	"github.com/sailpoint-oss/sailpoint-cli/internal/templates"
+	"github.com/sailpoint-oss/sailpoint-cli/internal/terminal"
+	"github.com/sailpoint-oss/sailpoint-cli/internal/types"
+	"github.com/spf13/cobra"
+)
+
+func newTemplateCmd() *cobra.Command {
+	var outputTypes []string
+	var folderPath string
+	var template string
+	cmd := &cobra.Command{
+		Use:     "template",
+		Short:   "generate a report using a template",
+		Long:    "generate a report from IdentityNow using a template",
+		Example: "sail report template",
+		Aliases: []string{"temp"},
+		Args:    cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			err := config.InitConfig()
+			if err != nil {
+				return err
+			}
+
+			apiClient, err := config.InitAPIClient()
+			if err != nil {
+				return err
+			}
+
+			var selectedTemplate templates.ReportTemplate
+			reportTemplates, err := templates.GetReportTemplates()
+			if err != nil {
+				return err
+			}
+
+			if len(args) > 0 {
+				template = args[0]
+			} else {
+				template, err = templates.SelectTemplate(reportTemplates)
+				if err != nil {
+					return err
+				}
+			}
+			if template == "" {
+				return fmt.Errorf("no template specified")
+			}
+
+			log.Log.Info("Selected Template", "template", template)
+
+			matches := types.Filter(reportTemplates, func(st templates.ReportTemplate) bool { return st.Name == template })
+			if len(matches) < 1 {
+				return fmt.Errorf("no template matches for %s", template)
+			} else if len(matches) > 1 {
+				log.Log.Warn("multiple template matches, the first match will be used", "template", template)
+			}
+			selectedTemplate = matches[0]
+			varCount := len(selectedTemplate.Variables)
+			if varCount > 0 {
```

Comment:
I don't think this check is needed, and just makes the code less readable than if you just add the len check in the for loop

Reply:
I believe you are spot on

Reply:
I was wrong!

This variable assignment exists because its used in the for loop directly below these lines